### PR TITLE
Make sure index variable comes first

### PR DESCRIPTION
**Pull request by bschreck**
Reorders variable list in `Entity.__init__()` so that the index variable is at the front of the list.

### CODE REVIEW

File: featuretools/tests/entityset_tests/test_entity.py [hunk 1]
```diff
@@ -22,6 +22,11 @@ def test_is_index_column(es):
     assert es['cohorts'].index == 'cohort'
 
 
+def test_index_at_beginning(es):
```
Comment (kmax12) [inline featuretools/tests/entityset_tests/test_entity.py]: can we add another test? 

I'm thinking create an entity from a DF where the index isn't the first column and make sure that it gets moved to be be the first in the variables list.

File: featuretools/entityset/entity.py [hunk 2]
```diff
@@ -120,6 +120,11 @@ def __init__(self, id, df, entityset, variable_types=None, name=None,
         if self.index is not None and self.index not in inferred_variable_types:
             self.add_variable(self.index, vtypes.Index)
 
+        # make sure index is at the beginning
+        index_variable = [v for v in self.variables
+                          if v.id == self.index][0]
+        self.variables = [index_variable] + [v for v in self.variables
```
Comment (kmax12) [inline featuretools/entityset/entity.py]: do we not care if the order of variable's doesn't match the underlying dataframe? I don't believe we assume that anywhere. can you think of any places?

Comment (bschreck) [reply featuretools/entityset/entity.py]: Why don't we reorder the columns in the dataframe as well?

File: featuretools/tests/entityset_tests/test_entity.py (hunk 1 above)
Comment (bschreck) [reply featuretools/tests/entityset_tests/test_entity.py]: Yeah makes sense, I'll add that too

File: featuretools/entityset/entity.py (hunk 2 above)
Comment (kmax12) [reply featuretools/entityset/entity.py]: ya, let's do that

**Comment by kmax12** [comment]
Looks good, merging

**Pull request merged**